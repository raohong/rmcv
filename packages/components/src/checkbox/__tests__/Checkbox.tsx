import { fireEvent, render, screen } from '@testing-library/react';
import { Checkbox, checkboxClassNames } from '..';

const testId = 'checkbox';

it('render correctly', () => {
  const tree = render(<Checkbox>name</Checkbox>);

  expect(tree.asFragment()).toMatchSnapshot();
});

it('render with checked', () => {
  render(<Checkbox data-testid={testId} checked />);

  expect(screen.getByTestId(testId)).toHaveClass(checkboxClassNames.checked);
});

it('render with value', () => {
  render(<Checkbox value='test' data-testid={testId} checked />);

  expect(
    screen.getByTestId(testId).querySelector(`.${checkboxClassNames.input}`),
  ).toHaveAttribute('value', 'test');
});

it('render with shape', () => {
  render(<Checkbox data-testid={testId} shape='square' />);

  expect(
    screen.getByTestId(testId).querySelector(`.${checkboxClassNames.inner}`),
  ).not.toHaveStyleRule('border-radius', '0.5em');
});

it('render with onChange', () => {
  const onChange = jest.fn();

  render(<Checkbox data-testid={testId} value={1} onChange={onChange} />);

  fireEvent.click(screen.getByTestId(testId));

  expect(onChange).toHaveBeenCalledWith(true);
});

it('render with disabled', () => {
  const onChange = jest.fn();

  render(<Checkbox data-testid={testId} onChange={onChange} disabled />);

  fireEvent.click(screen.getByTestId(testId));

  expect(screen.getByTestId(testId)).toHaveClass(checkboxClassNames.disabled);
  expect(onChange).not.toHaveBeenCalled();
});

it('render with labelPosition', () => {
  render(<Checkbox data-testid={testId} labelPosition='left' />);

  expect(screen.getByTestId(testId)).toHaveStyleRule(
    'flex-direction',
    'row-reverse',
  );
});

it('render with iconSize', () => {
  render(<Checkbox data-testid={testId} iconSize={18} checked />);

  expect(
    screen.getByTestId(testId).querySelector(`.${checkboxClassNames.inner}`),
  ).toHaveStyle({
    'font-size': '18px',
  });
});

it('render with checkedColor', () => {
  render(<Checkbox data-testid={testId} checkedColor='red' checked />);

  expect(
    screen.getByTestId(testId).querySelector(`.${checkboxClassNames.inner}`),
  ).toHaveStyleRule('border-color', 'red');
  expect(
    screen.getByTestId(testId).querySelector(`.${checkboxClassNames.inner}`),
  ).toHaveStyleRule('background-color', 'red');
});

it('render with renderIcon', () => {
  const com = render(
    <Checkbox
      data-testid={testId}
      renderIcon={checked => <span>{checked ? 'checked' : 'unchecked'}</span>}
      checked
    />,
  );

  expect(screen.getByText('checked')).toBeInTheDocument();

  com.rerender(
    <Checkbox
      data-testid={testId}
      renderIcon={checked => <span>{checked ? 'checked' : 'unchecked'}</span>}
      checked={false}
    />,
  );

  expect(screen.getByText('unchecked')).toBeInTheDocument();
});
