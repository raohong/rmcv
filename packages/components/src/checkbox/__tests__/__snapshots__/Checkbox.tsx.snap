// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-tap-highlight-color: transparent;
  -webkit-font-smooth: antialiased;
  position: relative;
  touch-action: none;
  list-style: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,Helvetica Neue,Helvetica,Segoe UI,Arial,Roboto,PingFang SC,miui,Hiragino Sans GB,Microsoft Yahei,sans-serif;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  gap: 8px;
}

.emotion-1 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 1em;
  min-width: 1em;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-size: 20px;
  box-sizing: content-box;
  transition-property: border-color,background-color,color;
  transition-duration: 0.2s;
  border: 1px solid #c8c9cc;
  border-radius: 0.5em;
}

.emotion-2 {
  position: absolute;
  left: 0;
  top: 0;
  right: 0;
  bottom: 0;
  opacity: 0;
  cursor: inherit;
  visibility: hidden;
}

.emotion-3 {
  color: #323233;
  line-height: 20px;
  box-sizing: border-box;
}

<label
    class="RVuiCheckbox-root emotion-0"
  >
    <div
      class="RVuiCheckbox-inner emotion-1"
    >
      <input
        class="RVuiCheckbox-input emotion-2"
        type="checkbox"
        value=""
      />
    </div>
    <span
      class="RVuiCheckbox-label emotion-3"
    >
      name
    </span>
  </label>
</DocumentFragment>
`;
